### PR TITLE
fix: updateImage and draw should not throw

### PR DESCRIPTION
**Pull request by mikehazell**
### Changes

This PR:
- removes the code which checks for a loaded image from the `updateImage` and `draw` methods.
- adds some tests to ensure `drawImageSync` is correctly verifying an elements state.
- fixes the condition in `drawImageSync` because of an edge case condition where the element has a layer added without an image.

### Some context

I'm not sure if this was the original intention of `updateImage` but it is often used to tell cornerstone that the **canvas state is invalid and needs to be re-rendered**.

There are several similar methods `drawImage`, `draw`, `invalidate`, `invalidateImageId` and `drawInvalidated`.

- All these end up setting one or both the flags `needsRedraw` and `invalid` on one or more `enabledElement`'s.
- None of these methods ever actually use any of the properties of the image.
- Only `draw` and `updateImage` currently have a check to see if the element has an image loaded.

### This check is not needed

`drawImageSync` is the method which actually causes something to be rendered to the canvas and is therefore the method which would encounter an error if some required state were missing. The implementation of this method already contains checks to ensure the `enabledElement` has all the state required to render and will quietly return early if it does not.

So if these checks are working correctly, there is no need for `draw` or `updateImage` to also check this state.

### Why is it worth changing this

The most obvious example of `updateImage` throwing causing pain is when a cornerstone tool is changing mode.

The tool may need to tell cornerstone that the state of the canvas needs to be updated. For example to stop showing a cursor, or to redraw without an annotation.

We can see a case of this causing a crashing bug here:
https://github.com/cornerstonejs/cornerstoneTools/issues/1303

This has also been an issue with the [BrushTool](https://github.com/cornerstonejs/cornerstoneTools/pull/925).

### CODE REVIEW

File: src/internal/drawImageSync.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default function (enabledElement, invalidated) {
   const layers = enabledElement.layers || [];
 
   // Check if enabledElement can be redrawn
-  if (!enabledElement.canvas || !(enabledElement.image || layers.length)) {
+  if (!enabledElement.canvas || !enabledElement.image) {
```
Comment (mikehazell) [inline src/internal/drawImageSync.js:24]: Checking the length of the layers array is not actually correct. It is absolutely possible and sometimes desirable to create a layer without an image. In this case we still cannot render.

Also, the first thing which will crash is on line 31 below where we're accessing the `stats` property of `enabledElement.image`.

Layers will set `elementElement.image` anytime the `activeLayer` is set which happens automagically when the first layer is added (assuming it has an image).

**Comment by mikehazell** [comment]
NOTE: I've removed `yarn.lock` because it was breaking the travis check. Happy to revert that change if you'd prefer but we really should have only one of `package-lock.json` and `yarn.lock`.

**Comment by mikehazell** [comment]
Hey @dannyrb, any thoughts on this?

**Review by zakjholt — APPROVED** [review]
This seems like it'll solve an issue I'm currently running into with the layers api

**Comment by swederik** [comment]
Thanks!
